src/server/services/toeristische-verhuur.ts

```diff
@@ -23,10 +39,86 @@ export interface ToeristischeVerhuurRegistratie {
   street: string | null;
 }
 
+export interface TransformedVakantieverhuur extends Vakantieverhuur {
+  isPast: boolean;
+  duration: number | null;
+}
+
+export interface TransformedVakantieverhuurAfmelding
+  extends VakantieverhuurAfmelding {
+  isPast: boolean;
+  duration: number | null;
+}
+
+export interface TransformedVakantieverhuurVergunningaanvraag
+  extends VakantieverhuurVergunningaanvraag {
+  isPast: boolean;
+  duration: number | null;
+}
+
+export interface TransformedBBVergunning extends BBVergunning {
+  isPast: boolean;
+  duration: number | null;
+}
+
+export type VakantieverhuurVergunningen =
+  | Vakantieverhuur
+  | VakantieverhuurAfmelding
+  | BBVergunning
+  | VakantieverhuurVergunningaanvraag;
+
+export type TransformedVakantieverhuurVergunningen =
+  | TransformedVakantieverhuur
+  | TransformedVakantieverhuurAfmelding
+  | TransformedBBVergunning
+  | TransformedVakantieverhuurVergunningaanvraag;
+
 export interface ToeristischeVerhuurRegistratiesSourceData {
   content: ToeristischeVerhuurRegistratie[];
 }
 
+export const daysLeftInCalendarYear = (items: Vakantieverhuur[]): number => {
+  const itemsThisYear = items?.filter((item) => {
+    return item.dateStart ? isCurrentYear(item.dateStart) : undefined;
+  });
+  return (
```

Comment:
Zo op het eerste gezicht ziet dit er niet uit alsof het in een reduce call gedaan moet worden. Kun je deze functionaliteit in een aparte functie zetten en  proberen het wat simpeler te schrijven? Mogelijk gebruik je ook de data waaraan je al de `duration` hebt toegevoevgd. 


iets als:
```
const totalDaysPlanned = verhuur.filter(verhuur => isThisYear(verhuur.dateStart)).map(verhuur => verhuur.duration).reduce((total, duration) => total+duration,0);
const daysLeft = MAX_DAYS - totalDaysPlanned;
```

Reply:
Voor readability zeker met je eens. Alleen is het niet inefficient om 3 keer er doorheen te loopen? Nou goed misschien is het gekloot in de marge hoor, echte performance problemen zal het niet opleveren.

Reply:
Readability is belangerijker in deze. Als we echte performance willen moet je reduce sowieso niet gebruiken maar gewoon `for () {}` gebruiken